languageserver/integration/flow.go

```diff
@@ -393,23 +393,37 @@ func (f *flowkitClient) createSigner(address flow.Address) (*accounts.Account, e
 }
 
 func (f *flowkitClient) GetCodeByName(name string) (string, error) {
-	contracts, err := f.state.DeploymentContractsByNetwork(config.EmulatorNetwork)
+	// Try to find the contract by name
+	contract, err := f.state.Contracts().ByName(name)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("couldn't find the contract by import identifier: %s", name)
 	}
 
-	for _, contract := range contracts {
-		if name == contract.Name {
-			return string(contract.Code()), nil
-		}
+	// If no location is set, return an error
+	if contract.Location == "" {
+		return "", fmt.Errorf("source file could not be found for import identifier: %s", name)
 	}
 
-	return "", fmt.Errorf(fmt.Sprintf("couldn't find the contract by import identifier: %s", name))
+	// Resolve the contract source code from file location
+	code, err := f.getCodeFromLocation(name, contract.Location)
+	if err != nil {
+		return "", err
+	}
+	return code, nil
 }
 
 // Helpers
 //
 
+// Helper function to get code from a source file location
+func (f *flowkitClient) getCodeFromLocation(name, location string) (string, error) {
+	code, err := f.loader.ReadFile(filepath.Join(filepath.Dir(f.getConfigPath()), location))
```

Comment:
Just for improved readability:
```suggestion
	dir := filepath.Dir(f.getConfigPath())
	path := filepath.Join(dir, location)
	code, err := f.loader.ReadFile(path)
```